clang/lib/CodeGen/Targets/DirectX.cpp

```diff
@@ -56,9 +75,18 @@ llvm::Type *DirectXTargetCodeGenInfo::getHLSLType(CodeGenModule &CGM,
 
     return llvm::TargetExtType::get(Ctx, TypeName, {ElemType}, Ints);
   }
-  case llvm::dxil::ResourceClass::CBuffer:
-    llvm_unreachable("dx.CBuffer handles are not implemented yet");
-    break;
+  case llvm::dxil::ResourceClass::CBuffer: {
+    QualType ContainedTy = ResType->getContainedType();
+    if (ContainedTy.isNull() || !ContainedTy->isStructureType())
```

Comment:
why would we return nullptr if the ContainedTy ia not struct? seems like we could return back a valid llvm::Type for this case?

Reply:
Contained type of the CBuffer resource handle is always a layout struct of the constant buffer (implicitly created by Sema [here](https://github.com/llvm/llvm-project/blob/e9fb2391210e20d1d4dcf8bbe37085df902ed028/clang/lib/Sema/SemaHLSL.cpp#L475)).

Reply:
Should this be an assert then?

Reply:
The pattern has been to return `nullptr` for types we don't know how to convert. An assert with message "Generic handling of HLSL types is not supported." will fire in `CGHLSLRuntime::convertHLSLSpecificType` if the converted type is null.